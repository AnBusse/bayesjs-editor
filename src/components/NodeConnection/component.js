import React from 'react';
import PropTypes from 'prop-types';

const NodeConnection = ({
  name,
  cx,
  cy,
  color,
}) => (
  <circle cx={cx} cy={cy} r="8" fill={color}>
    <title>{`Rede: ${name}`}</title>
  </circle>
);

NodeConnection.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  cx: PropTypes.number.isRequired,
  cy: PropTypes.number.isRequired,
};

export default NodeConnection;
