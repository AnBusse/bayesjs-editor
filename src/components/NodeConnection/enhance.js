import { withProps } from 'recompose';

const getCx = index => 75 + (20 * index);
const enhance = withProps(({ index }) => ({
  cx: getCx(index),
}));

export default enhance;
