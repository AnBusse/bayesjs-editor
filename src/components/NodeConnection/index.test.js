import React from 'react';
import { shallow } from 'enzyme';
import NodeConnection from './component';

const defaultProps = {
  name: 'name',
  cx: 0,
  cy: 0,
  color: '#ddd',
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<NodeConnection {...compProps} />);
};

describe('NodeConnection Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });
});
