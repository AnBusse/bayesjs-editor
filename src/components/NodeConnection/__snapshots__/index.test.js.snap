// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeConnection Component matches snapshot 1`] = `
<circle
  cx={0}
  cy={0}
  fill="#ddd"
  r="8"
>
  <title>
    Rede: name
  </title>
</circle>
`;
